src/extension.ts

```diff
@@ -197,7 +197,7 @@ async function activateServerForFolder(context: ExtensionContext, uri: Uri, fold
   }
 
   let extraArgs: string = workspace.getConfiguration('haskell', uri).serverExtraArgs;
-  if (extraArgs !== null) {
+  if (extraArgs !== '') {
```

Comment:
1. Are you sure the deletion of the null check will not cause a null dereference at `extraArgs.split`?
2. What's the point of checking for empty strings if you filter them out at line 218?